### ISSUE TITLE
Transient failure accessing nugets redirects CoreCLR build to a broken backup link

### ISSUE THREAD

**Issue by dingydandy** [dotnet/runtime#42370]
*Initial cost estimate*: 2 days
*Initial contacts*: @trylek, @ViktorHofer 

I get coreclr source code from https://github.com/dotnet/runtime.git, the master branch, and I want to biuld the source code, so I run build.cmd, however, I get the following feedback:
E:\Code\netcore src\coreclr>.\build.cmd
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-sdk-5.0.100-rc.1.20454.5-win-x64.zip
dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-sdk-5.0.100-rc.1.20454.5-win-x64.zip
dotnet-install: Downloading legacy link: https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-dev-win-x64.5.0.100-rc.1.20454.5.zip
dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-dev-win-x64.5.0.100-rc.1.20454.5.zip
Failed to install dotnet from public location.
Some builds failed:
        Configuration: Debug, Architecture: x64

So, what's wrong, can anyone tell me how to build the coreclr cource code? I need help.

[trylek] Based on below discussion I believe that the actionable work item here is making sure we don't resort to a broken backup link when the main nuget link is broken and instead we display a sufficiently descriptive error message.

**Comment by EgorBo** [comment]
Are you under some proxy?

**Comment by dingydandy** [comment]
@EgorBo No, I tried without proxy, still not work, however, when I paste the url https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-dev-win-x64.5.0.100-rc.1.20454.5.zip, I get the following errors: 
![image](https://user-images.githubusercontent.com/33050122/93867929-4ce11900-fcfc-11ea-97b7-d29d25ddbe00.png)
it seams that the blob does not exists

**Comment by trylek** [comment]
Hmm, just tried that and now it seems to be working for me; to be precise, I'm not hitting the first download failure so I'm not observing runtime build resorting to the fallback link that is indeed broken. @dingydandy - in your previous response you demonstrated that the "fallback link" doesn't work (I concur, it doesn't work for me either); but the primary link does work for me - can you please clarify whether the primary link,

https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-sdk-5.0.100-rc.1.20454.5-win-x64.zip

is still inaccessible for you or whether that was just a transient issue (so that you're now unblocked w.r.t. running "build" in your clone of the runtime repo)? @ViktorHofer, have you got any idea where the malfunctioning backup link,

https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20454.5/dotnet-dev-win-x64.5.0.100-rc.1.20454.5.zip

is coming from? I don't see the value in resorting to a broken backup link if the primary link doesn't work but I have no idea where this should be fixed.

@dingydandy - just as a heads-up, the snippet in the issue title seems to indicate that the folder with your clone of the runtime repo is on a path containing spaces; a new issue was filed last week by someone who found out that parts of runtime build got broken in this scenario (it worked several months back but has apparently bit-rotten):

https://github.com/dotnet/runtime/issues/42397

Thanks

Tomas

**Comment by trylek** [comment]
@dingydandy - Hmm, rereading the issue description I think I may have slightly misunderstood the steps - upon a second glance it looks like that your location of the runtime repo clone is actually <code>E:\Code\netcore</code> and you're trying to run <code>build.cmd</code> in the folder <code>src\coreclr</code> - but that script is no longer there, the only remaining script there is <code>build-runtime.cmd</code> as CoreCLR is supposed to be built using the root build script i.e. something like

<pre>
E:\netcore>build clr
</pre>

Can you please double-check how old runtime repo clone you're using and / or retry with a fresh clone of the runtime repo in a different folder? Alternatively, please double-check whether you don't have stale artifacts in the repo clone or use <code>git clean -xdf</code> to get rid of them.

**Comment by ViktorHofer** [comment]
Closing in favor of https://github.com/dotnet/runtime/issues/34015 which tracks improving the SDK bootstrap scripts.
